### DISCUSSION TITLE
modelgrid class (possible bug?)

### DISCUSSION THREAD

**Discussion by d-strom** [modflowpy/flopy#1996] [Q&A] [answered]
I apologize in advance for starting another thread on a possibly already answered topic. 

I still dont quite understand why the modelgrid function doesnt work for me...

I have been looking at the example i the image below.

![image](https://github.com/modflowpy/flopy/assets/74509984/5d36684d-10b9-467a-9675-9d3440ae547a)

I tried to use the same procedure for my own data, but for whatever reason the modelgrid returns an attribute error.

![image](https://github.com/modflowpy/flopy/assets/74509984/6dc4aa07-ae27-4d23-8826-c5263e1ada83)

using the following line of code I have atleast been able to access the grid information:

sr = flopy.discretization.vertexgrid.VertexGrid.from_binary_grid_file(grb_name, verbose=False)

Why cant I access my model with the modelgrid class?

Sincerely
David

**Comment by wpbonelli** [comment]
@DStrom1987 check the `sim.model_names` property to see if a model "mod0000" really exists in your simulation?

**Reply by d-strom** [reply]
![image](https://github.com/modflowpy/flopy/assets/74509984/f42a48c0-4d9e-4ac8-9f43-8833f1968f4c)

But there clearly has been a model with that name imported?

![image](https://github.com/modflowpy/flopy/assets/74509984/cc9d87ce-3aa9-4851-abd3-f3577fac418e)

**Reply by wpbonelli** [reply]
Flopy returns `None` from `get_model()` if a model with the given name is not found https://github.com/modflowpy/flopy/discussions/1992#discussioncomment-7423873. Maybe there is a case to be made for e.g. a `strict` boolean parameter causing it to raise an error if the requested model isn't found, however.

**Reply by d-strom** [reply]
hmm, it does not return None.

Same goes for another case.

![image](https://github.com/modflowpy/flopy/assets/74509984/e8337b8e-b994-4397-9a93-49a1c1100fdf)

![image](https://github.com/modflowpy/flopy/assets/74509984/abd2fe60-f135-4a20-a9fe-b4879bb64e4a)

![image](https://github.com/modflowpy/flopy/assets/74509984/f79e5dea-16a0-441a-a33e-3d8e4c732e47)

**Reply by wpbonelli** [reply]
`MFSimulation` does not have a `modelgrid` property, only the model. In your case `vertex_sim` is a simulation, not a model.

**Reply by d-strom** [reply]
my bad. Still, same happens if i try and load in the vertex_m (i.e. the model) in such manner as in the example.

edit: turns out print(vertex_m) returns None after all.. Dont really know why? All that is going on is 
sim = flopy.mf6.MFSimulation.load(sim_ws=tmp_d, verbosity_level=0) and m = sim.getmodel("modelname")

**Reply by wpbonelli** [reply, marked as answer]
It looks like your model name is "modflow", try using that name. If you only have one model in your simulation, you can also use `get_model()` without providing a name.

**Reply by d-strom** [reply]
that did it! How strange.. The models name is disv as far as I can tell.  Which is what I named it. Anyway, thank you. At least it worked!

![image](https://github.com/modflowpy/flopy/assets/74509984/7760fd47-f32d-4933-a1d5-e7caad9c8934)
